### ISSUE TITLE
Old FR algorithm (maybe KK, too?)

### ISSUE THREAD

**Issue by gaborcsardi** [igraph/rigraph#137]
Maybe in another package, actually.

**Comment by mbojan** [comment]
+1 for re-adding old algs as separate functions.

> Maybe in another package, actually.

Why? What's with this package-splitting frenzy? These are all functions for the same purpose and tightly related to igraph data representation.

**Comment by gaborcsardi** [comment]
> Why? What's with this package-splitting frenzy? These are all functions for the same purpose and tightly related to igraph data representation.

Two reasons.

One. It is a pain to maintain a package of this size.

Two. It is also a pain to depend on it. Often you just need a small functionality of igraph, and currently you need to install the whole huge thing to get that.

**Comment by mbojan** [comment]
> One. It is a pain to maintain a package of this size.

Is it really easier to maintain large number of tiny packages rather than smaller number of larger packages of related functionality? I think of overheads like CRAN submissions, documentation consistency etc.

> Two. It is also a pain to depend on it. Often you just need a small functionality of igraph, and currently you need to install the whole huge thing to get that.

In case of igraph I can think of projects, in which, say, I only need igraph class for graph data representation without using any visualization functions or analysis functions. I dare to say such projects are probably a minority as compared to projects in which one needs all three things (data, vis, analysis). If all is split into several packages, then indeed this will be a dependency pain as I will have to load several packages instead of one.

Package developers can anyway carve their NAMESPACEs carefully with selective imports/exports.

Don't you agree?

I'm writing this as a dedicated igraph user and popularizer out of concern what are the plans for igraph future development.

**Comment by gaborcsardi** [comment]
> Is it really easier to maintain large number of tiny packages rather than smaller number of larger packages of related functionality? 

Yes, it is. Does not have to be a large number, actually.

> Package developers can anyway carve their NAMESPACEs carefully with selective imports/exports.

Sure, but they still need to install igraph. Often you just need one or two small functions from igraph, e.g. a topological sort. I needed that in three different projects in the last month.

Anyway, there should be a good way of adding extension package to igraph, that depend on the C code (as well).

**Comment by mbojan** [comment]
I'm not sure I'm totally convinced, but OK. Definitely it takes a lot of time to build igraph currently, so small additions can be pain to run `R CMD check`, tests and so on.

BTW I'm curious about the new R API in igraph: new function names, e.g. `with_*`, `layout_with_*` etc. I would be very happy to learn what are your plans in that respect. I teach some courses/tutorials with igraph so some heads-up would be very beneficial in case I need to do some updates to the materials. Perhaps we could go off GitHub to michal2992@gmail.com? Thanks in advance.

**Comment by gaborcsardi** [comment]
No big plans now. The old function names were a mess, new ones are better. That's all.

**Comment by ko-ichi-h** [comment]
Yes, that's very nice! I desperately want to use older (0.7.0) FR & KK layout functions!
